src/core/View.js

```diff
@@ -525,6 +521,71 @@ export default class View extends Component {
     this.renderWindow.render();
   }
 
+  pickClosest(xp, yp, tolerance) {
+    const x1 = Math.floor(xp - tolerance);
+    const y1 = Math.floor(yp - tolerance);
+    const x2 = Math.ceil(xp + tolerance);
+    const y2 = Math.ceil(yp + tolerance);
+
+    this.selector.setArea(x1, y1, x2, y2);
+    this.previousSelectedData = null;
+
+    if (this.selector.captureBuffers()) {
+      const pos = [xp, yp];
+      const maxDistance = tolerance;
+      const outSelectedPosition = [0, 0];
+      const info = this.selector.getPixelInformation(
```

Comment:
Is that the magic line that will deal with the tolerance?

Reply:
Yes, the `getPixelInformation` function, which was already present in `vtk-js / vtkOpenGLHardwareSelector`, does a search from pointer position outward within the tolerance zone. Previously `generateSelection` function was used which returns an array of selections within the tolerance footprint. for `pick` and `hover` operations we need the closest actor. Not sure if this generalizes well for other applications!?

Reply:
Maybe we want to pass the expected tolerance. if 0, then it must be the pixel right below the ray.

Reply:
Isn't tolerance specified by `maxDistance` here? Or am I missing something?

Reply:
Sure thing.